src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -257,133 +263,201 @@ private void enqueueNotification(TripMonitorNotification ...tripMonitorNotificat
-        //  value. Also, we may want to use the start time leading up to the trip (for periodic monitoring) and once
-        //  the trip has started, switch to some other monitoring interval while the trip is in progress.
-        ZonedDateTime tripTime = ZonedDateTime.ofInstant(trip.itinerary.startTime.toInstant(), zoneId)
-            // Offset the trip time to today's date.
-            .withYear(now.getYear())
-            .withMonth(now.getMonthValue())
-            .withDayOfMonth(now.getDayOfMonth());
+        ZonedDateTime now = DateTimeUtils.nowAsZonedDateTime(targetZoneId);
         // TODO: Change log level.
-        LOG.info("Trip {} starts at {} (now={})", trip.id, tripTime.toString(), now.toString());
-        // TODO: This check may eventually make use of endTime and be used in conjunction with tripInProgress check (see
-        //  above).
-        boolean tripHasEnded = now.isAfter(tripTime);
-        // Skip check if trip is not active today.
-        if (!trip.isActiveOnDate(now)) {
-            // TODO: Change log level.
-            LOG.info("Trip {} not active today.", trip.id);
-            return true;
-        }
-        // If the trip has already occurred, clear the journey state.
-        // TODO: Decide whether this should happen at some other time.
-        if (tripHasEnded) {
-            // TODO: Change log level.
-            LOG.info("Trip {} has cleared.", trip.id);
-            // TODO: If we decide to stack up responses, we should clear them here.
-//            trip.clearJourneyState();
-            return true;
-        }
+        LOG.info("Trip starts at {} (now={})", tripStartInstant.toString(), now.toString());
+
         // If last check was more than an hour ago and trip doesn't occur until an hour from now, check trip.
-        long millisSinceLastCheck = DateTimeUtils.currentTimeMillis() - trip.retrieveJourneyState().lastChecked;
+        long millisSinceLastCheck = DateTimeUtils.currentTimeMillis() - journeyState.lastCheckedMillis;
         long minutesSinceLastCheck = TimeUnit.MILLISECONDS.toMinutes(millisSinceLastCheck);
-        LOG.info("{} minutes since last check", minutesSinceLastCheck);
-        long minutesUntilTrip = Duration.between(now, tripTime).toMinutes();
-        LOG.info("Trip {} starts in {} minutes", trip.id, minutesUntilTrip);
-        // TODO: Refine these frequency intervals for monitor checks.
+        LOG.info("{} minutes since last checking trip", minutesSinceLastCheck);
+        long minutesUntilTrip = (tripStartInstant.getEpochSecond() - now.toEpochSecond()) / 60;
+        LOG.info("Trip starts in {} minutes", minutesUntilTrip);
+        // skip check if the time until the next trip starts is longer than the requested lead time
+        if (minutesUntilTrip > trip.leadTimeInMinutes) {
+            LOG.info(
+                "Next trip start time is greater than lead time of {} minute(s). Skipping trip.",
```

Comment:
Tweak log text... maybe "The time until this trip begins again is more than the {}-minute lead time."?